src/lib/GoogleTransit.js

```diff
@@ -0,0 +1,84 @@
+import rp from 'request-promise';
+import moment from 'moment';
+
+
+export default class GoogleTransit {
+  constructor(origin, destination, departTime) {
+    this.origin = origin;
+    this.destination = destination;
+    this.departTime = departTime;
+  }
+
+  _getDepartTimeUnixEpoch () {
+    let now = moment().valueOf();
+    let departTimeToday = moment().startOf('day').add(this.departTime, 'h').valueOf()/1000;
+    let departTimeTomorrow = moment().startOf('day').add(this.departTime + 24, 'h').valueOf()/1000;
+    if (now > departTimeToday) {
+      return departTimeTomorrow;
+    } else {
+      return departTimeToday;
+      }
+  }
+  async getData() {
+    const normal = await this._getDataDepartNormal();
+    const now = await this._getDataDepartNow();
+    return {
+      distance: normal.routes[0].legs[0].distance.text,
+      durationNow: now.routes[0].legs[0].duration.text,
+      arrivalNow: now.routes[0].legs[0].arrival_time.text,
+      durationNormal: normal.routes[0].legs[0].duration.text,
+      arrivalNormal: normal.routes[0].legs[0].arrival_time.text,
+    };
+  }
+  async _getDataDepartNormal() {
+    try {
+      let data = await rp({
+        method: "GET",
+        url: `https://maps.googleapis.com/maps/api/directions/json?` +
+          `origin=${this.origin}&destination=${this.destination}&` +
+          `departure_time=${this._getDepartTimeUnixEpoch()}&mode=transit` +
+          `&key=AIzaSyCyVBmd0JigVvVazx5O6e3OHwopFn0tklY`,
```

Comment:
a better way to do this is with the `qs` option: 
```
  qs : {
    origin: this.origin,
    departure_time: this._getDepartTimeUnixEpoch(),
    mode: 'transit',
    key: process.env.GOOGLE_API_KEY
  }
```
Also you should keep your google API key as an environment variable in .env rather than committing it. good practice to get into for security reasons and makes it easier to have a separate key for development/production or to allow people to create their own projects without modifying your code.